Common/Internal/FastUtf8JsonWriter.cs

```diff
@@ -18,18 +18,20 @@ internal class FastUtf8JsonWriter : IFastJsonWriter {
         private byte[] _buffer;
         private readonly Encoder _encoder;
 
+        private int _stateStackIndex;
         private readonly ArrayPool<State> _stateStackPool;
         private readonly State[] _stateStack;
-        private int _stateStackIndex = 0;
 
         private FastUtf8JsonStringWriter? _stringWriter;
 
         public FastUtf8JsonWriter(ArrayPool<byte> bufferPool) {
             _bufferPool = bufferPool;
             _buffer = bufferPool.Rent(4096);
 
+            _stateStackIndex = 0;
             _stateStackPool = ArrayPool<State>.Shared;
             _stateStack = _stateStackPool.Rent(64);
+            _stateStack[0] = State.None;

```

Comment:
Is this the bug fix?

Reply:
Yep, since the memory you get from pool is not guaranteed to be all 0s.
States at `[1]` and so on are all fine as `PushState` always sets the state explicitly when moving the index.

I tried to start with `_stateStackIndex == -1` and `if` in `GetState`, but that seemed slower than this as `GetState` is called more often that the constructor (though I am having trouble measuring performance reliably).